### PR TITLE
Convert project to an arduino project

### PR DESCRIPTION
**Pull request by WillB97**
Can now be built and uploaded with the Arduino IDE.

### CODE REVIEW

File: .github/workflows/compile.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+on:
+  - push
+  - pull_request
+jobs:
+  compile-sketch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (RealOrangeOne) [inline .github/workflows/compile.yml]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/compile.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      - uses: arduino/compile-sketches@v1
+        with:
+          fqbn: "arduino:avr:uno"
+          sketch-paths: |
+            - src
```
Comment (RealOrangeOne) [inline .github/workflows/compile.yml]: Thought: Is it worth adding the built firmware as an artifact?

File: test_fw/src/ruggeduino_test.cpp
Comment (RealOrangeOne) [inline test_fw/src/ruggeduino_test.cpp:1]: Thought: I'm not sure how I feel about deleting these test files. From the looks of the code, right might have originated elsewhere?

File: .github/workflows/compile.yml (hunk 2 above)
Comment (WillB97) [reply .github/workflows/compile.yml]: Not really the IDE will build it before flashing, this just proves it will build.

**Review by RealOrangeOne — APPROVED** [review]

**Pull request merged**